### ISSUE TITLE
Ship the bounding box with the datasets when the reader supports it

### ISSUE THREAD

**Issue by mraspaud** [pytroll/satpy#933]
## Feature Request

The bounding box of datasets could be used more, for example in the dynamic computation of areas just covering the dataset. (Mostly for polar data as I think of it now)

At the moment, we just have a couple of readers extracting the bounding box of dataset, and they are not attached to the datasets themselves. 

By providing the bounding box as an attribute to the dataset, later processes will be able to make use of it for some operations.

**Comment by djhoese** [comment]
Right now we use SwathDefinitions for geolocation information. Would it be better to make the SwathDefinition able to accepting the bounding box information on `init`? Otherwise, it could compute it.

Another option for better bounding box calculations would be to chunk the lon/lat arrays specially so the outer edges and middle parts are all their own chunks.

**Comment by mraspaud** [comment]
Shipping it with the SwathDefinition could work, but would need satpy to handle it specially and pass it. No against it, just saying :)

Regarding special chunks sizes, I'm a bit skeptical, especially if the lons and lats need to be interpolated. Or alternatively we could have original data and interpolated data in different chunks?

**Comment by mraspaud** [comment]
So maybe this can be related to this issue: https://github.com/pytroll/python-geotiepoints/issues/18

**Comment by mraspaud** [comment]
And maybe also to the handling of GCPs ?

**Comment by djhoese** [comment]
Yeah, playing with chunk sizes is iffy, you're right. My thought with the information in the SwathDefinition was that pyresample could add the functionality to SwathDefinition and to the omerc calculation stuff but whether or not Satpy readers take advantage of it is a separate process. SwathDefinition could have it implemented as a method or property that computes them if they don't exist. This is pyresample 2.0 kind of stuff perhaps.

I'd also like to avoid things being in SwathDefinition that don't have a pure xarray equivalent. In this case I think the bounding box information could be an attribute of the longitude and latitude coordinates (ex. `scn['myds'].coords['longitude'].attrs['bounding_box']`).

**Comment by mraspaud** [comment]
The bounding box actually applies to any dataset in principle, not just only lons/lats, so `myds` should also have this `bounding_box` attribute :)

(side question: are the coords as dask arrays possible now, without being computed ?)
